tests/func/test_data_status.py

```diff
@@ -423,6 +427,29 @@ def test_missing_remote_cache(M, tmp_dir, dvc, scm, local_remote):
     }
 
 
+def test_not_in_remote_respects_not_pushable(
+    tmp_dir: TmpDir, dvc: Repo, scm, mocker, local_remote
+):
+    stages: list[Stage] = tmp_dir.dvc_gen({"foo": "foo", "dir": {"foobar": "foobar"}})
```

Comment:
`dvc.stage.create(outs=["foo"], single_stage=True)` did not generate `hash_info`, so I moved everything into `dvc_gen`. (We anyways want to generate the dir and pushable file).